### PR TITLE
Migration to Go Modules

### PR DESCRIPTION
**Pull request by UnderGreen**
https://github.com/gravitational/gravity/pull/1178 is stale branch now and I don't want to rebase from it. 
Everything the same as in PR above + I bumped all dependencies which were changed after that PR.

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 ENV GRPC_GATEWAY_ROOT /src/github.com/grpc-ecosystem/grpc-gateway
 ENV GOGOPROTO_ROOT /src/github.com/gogo/protobuf
 ENV PROTOC_URL https://github.com/google/protobuf/releases/download/v${PROTOC_VER}/protoc-${PROTOC_VER}-${PROTOC_PLATFORM}.zip
 
-RUN adduser jenkins --uid=995 --disabled-password --system
-RUN (mkdir -p /gopath/src/github.com/gravitational/gravity && \
-     chown -R jenkins /gopath && \
+RUN (groupadd jenkins --gid=$GID -o && \
+     useradd jenkins --uid=$UID --gid=$GID --create-home --shell=/bin/sh && \
+     # install some development libraries used when compiling fio
```
Comment (a-palchikov) [inline build.assets/Dockerfile:18]: nit: can you remove ~~some~~?

File: build.assets/Dockerfile [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
 ARG PROTOC_PLATFORM
 ARG GOGO_PROTO_TAG
 ARG GRPC_GATEWAY_TAG
-ARG GODEP_TAG
 ARG VERSION_TAG
+ARG UID
+ARG GID
 
 ENV TARBALL protoc-${PROTOC_VER}-${PROTOC_PLATFORM}.zip
 ENV GRPC_GATEWAY_ROOT /src/github.com/grpc-ecosystem/grpc-gateway
```
Comment (a-palchikov) [inline build.assets/Dockerfile:12]: Need to prefix those with `/gopath` as well.

File: build.assets/Makefile [hunk 3]
```diff
@@ -335,13 +336,13 @@ $(GOCACHE):
 .PHONY: $(TARGETS)
 $(TARGETS): | $(GOCACHE)
 	@echo -e "\n----> Building $@ binary...\n"
-	go build -o $@ -ldflags $(GRAVITY_LINKFLAGS) $(GRAVITY_BUILDFLAGS) -tags "$(GRAVITY_BUILDTAGS)" $(GRAVITY_PKG_PATH)/tool/$(@F)
+	GO111MODULE=on GOMOD=$(SRCDIR)/go.mod go build -mod=vendor -o $@ -ldflags $(GRAVITY_LINKFLAGS) $(GRAVITY_BUILDFLAGS) -tags "$(GRAVITY_BUILDTAGS)" $(GRAVITY_PKG_PATH)/tool/$(@F)
```
Comment (a-palchikov) [inline build.assets/Makefile:339]: Hmm, GOMOD is one of the environment variables that's being synthesized but not read:
```
$ go help environment
...
Additional information available from 'go env' but not read from the environment:
	...
	GOMOD
		The absolute path to the go.mod of the main module,
		or the empty string if not using modules.
```

Comment (UnderGreen) [reply build.assets/Makefile:339]: I am not sure I followed what did you mean. BTW, when I deleted GOMOD from the line a build is still working, but I believe in my previous PR it was not working. :man_shrugging:

Comment (a-palchikov) [reply build.assets/Makefile]: I mean go tools do not read from `GOMOD` but simply populate it to show which go.mod file is in use.

File: Makefile [hunk 4]
```diff
@@ -223,7 +221,7 @@ build:
 # 'install' uses the host's Golang to place output into $GOPATH/bin
 .PHONY: install
 install:
-	go install -ldflags $(GRAVITY_LINKFLAGS) -tags "$(GRAVITY_BUILDTAGS)" ./tool/tele ./tool/gravity
+	GO111MODULE=on go install -ldflags $(GRAVITY_LINKFLAGS) -tags "$(GRAVITY_BUILDTAGS)" ./tool/tele ./tool/gravity
```
Comment (a-palchikov) [inline Makefile:224]: This will probably need the `-mod=vendor` too?

Comment (UnderGreen) [reply Makefile:224]: Right. Will add it.

File: go.mod [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+	github.com/mitchellh/copystructure v1.0.0 // indirect
+	github.com/mitchellh/go-ps v1.0.0
+	github.com/mitchellh/hashstructure v1.0.0 // indirect
+	github.com/mreiferson/go-httpclient v0.0.0-20160630210159-31f0106b4474 // indirect
+	github.com/nsf/termbox-go v0.0.0-20190325093121-288510b9734e // indirect
+	github.com/olekukonko/tablewriter v0.0.1
+	github.com/opencontainers/go-digest v1.0.0-rc1
+	github.com/opencontainers/image-spec v1.0.1 // indirect
+	github.com/opencontainers/runc v0.1.1 // indirect
+	github.com/opencontainers/selinux v1.3.0
```
Comment (a-palchikov) [inline go.mod:113]: `v1.4.0` to keep it consistent with planet

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+	github.com/xtgo/set v1.0.0
+	github.com/zclconf/go-cty v0.0.0-20180829180805-c2393a5d54f2 // indirect
+	github.com/ziutek/mymysql v1.5.4 // indirect
+	go.mongodb.org/mongo-driver v1.0.4 // indirect
+	golang.org/x/crypto v0.0.0-20191011191535-87dc89f01550
+	golang.org/x/net v0.0.0-20190812203447-cdfb69ac37fc
+	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e // indirect
+	golang.org/x/sys v0.0.0-20191010194322-b09406accb47
+	gonum.org/v1/gonum v0.6.1 // indirect
+	google.golang.org/grpc v1.20.1
```
Comment (a-palchikov) [inline go.mod:146]: `v1.26.0` to keep consistent with planet.

**Review by a-palchikov — APPROVED** [review]

**Pull request merged**